client.go

```diff
@@ -858,3 +888,77 @@ func (c *Client) SetMaxBytesToParse(max int) *Client {
 	c.maxBytesToParse = max
 	return c
 }
+
+// getIpFromInterface return a IPv4 and IPv6 address from the interface.
+// if there is no address than that IPfamily is nil.
+func getIpFromInterface(theInterface string) (theIPs localTCPIP) {
+	if theInterface == "" {
+		return
+	}
+	ifaces, err := netInterfaces()
+	if err != nil {
+		log.Errorf("getIpFromInterface: %+v\n", err.Error())
+		return
+	}
+	log.Debugf("netInterfaces: %+v\n", ifaces)
+	for _, i := range ifaces {
+		if i.Name != theInterface {
+			continue
+		}
+		addrs, err := netInterfaceAddrs(&i)
+		if err != nil {
+			log.Errorf("getIpFromInterface: %+v\n", err.Error())
+			continue
+		}
+		for _, a := range addrs {
+			log.Debugf("Addr : %+v\n", a)
+			switch v := a.(type) {
+			case *net.IPAddr:
+				log.Debugf("%v : %s (%s)\n", i.Name, v, v.IP.DefaultMask())
+				if version4, tcpAddr := isIPv4(v.IP.String()); version4 != nil {
+					if *version4 {
+						theIPs.IPv4 = tcpAddr
+					} else {
+						theIPs.IPv6 = tcpAddr
+					}
+				}
+			case *net.IPNet:
+				log.Debugf("IPNET %v : %s (%s)\n", i.Name, v, v.IP.DefaultMask())
+				if version4, tcpAddr := isIPv4(v.IP.String()); version4 != nil {
+					if *version4 {
+						theIPs.IPv4 = tcpAddr
+					} else {
+						theIPs.IPv6 = tcpAddr
+					}
+				}
+			}
+		}
+	}
+	return
+}
+
+func isIPv4(ip string) (*bool, *net.TCPAddr) {
```

Comment:
Why pointer to bool instead of plain bool? I have not disassembled the resulting binary, but sounds to be quite a bit of overhead.